packages/app/src/app/overmind/namespaces/editor/internalActions.ts

```diff
@@ -338,7 +322,27 @@ export const forkSandbox: AsyncAction<{
       });
     }
 
-    await actions.internal.setCurrentSandbox(forkedSandbox);
+    // When we have a server we want to set a brand new sandbox,
+    // as the preview also needs to be managed correctly
+    if (getTemplateDefinition(forkedSandbox.template).isServer) {
+      await effects.vscode.closeAllTabs();
+      actions.internal.setCurrentSandbox(forkedSandbox);
```

Comment:
It's interesting, from a first glance this seems to imply that some more is happening in this function. Might be a good thing in the future to rename this or something, not now though 😄

Reply:
Yeah, lets save this for the refactor after components are ready. I will refactor the namespaces and the actions 😄